src/wallet/wallet_list.c

```diff
@@ -163,22 +170,49 @@ bool get_wallet_data_by_id(const uint8_t *wallet_id, Wallet *wallet) {
     }
   }
 
+  // If control reaches here, that means `wallet_id` search failed
+  if (reject_cb) {
+    reject_cb(ERROR_COMMON_ERROR_WALLET_NOT_FOUND_TAG, 0);
+  }
+
   return false;
 }
 
-bool get_wallet_name_by_id(const uint8_t *wallet_id, uint8_t *wallet_name) {
+bool get_wallet_name_by_id(const uint8_t *wallet_id,
+                           uint8_t *wallet_name,
+                           rejection_cb *reject_cb) {
   if (NULL == wallet_id) {
     return false;
   }
 
-  Wallet wallet = {0};
-  if (false == get_wallet_data_by_id(wallet_id, &wallet)) {
-    return false;
+  for (uint8_t wallet_idx = 0; wallet_idx < MAX_WALLETS_ALLOWED; wallet_idx++) {
+    wallet_state state = INVALID_WALLET;
+    if (wallet_is_filled(wallet_idx, &state)) {
+      continue;
+    }
+
+    if (0 == memcmp(wallet_id, get_wallet_id(wallet_idx), WALLET_ID_SIZE)) {
+      // Check if wallet is in usable state or not
+      if (INVALID_WALLET == state || is_wallet_locked(wallet_idx) ||
+          is_wallet_partial(wallet_idx) || is_wallet_unverified(wallet_idx) ||
+          is_wallet_share_not_present(wallet_idx)) {
+        if (reject_cb) {
+          reject_cb(ERROR_COMMON_ERROR_WALLET_PARTIAL_STATE_TAG, 0);
+        }
+        return false;
+      } else {
+        if (wallet_name) {
+          memcpy(wallet_name, get_wallet_name(wallet_idx), NAME_SIZE);
+        }
+        return true;
+      }
+    }
```

Comment:
This looks like logic duplication, consider using an implementation similar to previous one where we call `get_wallet_data_by_id` to fetch name.

Reply:
Got it, updated